lumberdev/sum-it-up
# feat: combine readability and openai api into one easy POST api

## WHAT IT DOES:

_Description of the pull request, what changed_
Turned Rhut's API into a POST api that fetches data from client, parses, and returns content from openAI sdk.

Made a new client side fetch and roughly hooked up all the client inputs to call it.


## HOW TO TEST:

_Description of how to test if applicable_

Do a post request with this data to `/api/fetchSummary` route (be careful)
with a body similar to this (for song data)
```json
{
  "type": "song",
  "url":"https://www.google.com/search?q=rhiannon+lyrics&rlz=1C5CHFA_enUS997US997&oq=rhiannon+lyrics&aqs=chrome..69i57j0i512l9.2440j0j7&sourceid=chrome&ie=UTF-8",
  "wordLimit": 100
}
```

for article data:
```json
{
  "type": "article",
  "url":"https://medium.com/dartlang/dart-3-alpha-f1458fb9d232",
  "wordLimit": 100
}
```

for text input data:
```json
{
  "type": "article",
  "text":"Here's some cool text, please make sure to enter enough text here! This is how you enter text. First type some letters on the keyboard, and there you have it. You've entered some text. Neat!",
  "wordLimit": 50
}
```
Make sure to have these keys set in your env
```
OPENAI_API_KEY
```

### TASK:

### WHAT'S LEFT:

_Delete if not necessary:_

- [ ] - gotta authenticate that route because this is a public repo.
- [ ] - need to clean this up, the code got messy
- [ ] - pass a variable to set up the correct api route
